k9mail-library/src/main/java/com/fsck/k9/mail/Message.java

```diff
@@ -57,7 +57,7 @@ public int hashCode() {
         final int MULTIPLIER = 31;
 
         int result = 1;
-        result = MULTIPLIER * result + mFolder.getName().hashCode();
+        result = MULTIPLIER * result + (mFolder != null ? mFolder.getName().hashCode() : 0);
```

Comment:
Can `mFolder == null` happen during regular operations? Or is it only so we can be lazy in tests?

Reply:
Little bit of both. A typical newly constructed MimeMessage instance won't have a folder yet, and iirc there isn't even a way to set one in MimeMessage itself.